File: python_solutions/chapter_05_bit_manipulation/problem_05_06_conversion.py
def conversion(A, B):
    a = A
    b = B
    bits_to_shift = 0
    while a > 0 or b > 0:
        if a & 1 == 1 and b & 1 == 0:
            bits_to_shift += 1
        elif b & 1 == 1 and a & 1 == 0:
            bits_to_shift += 1
        a >>= 1
        b >>= 1
    return bits_to_shift

File: cpp_solutions/chapter_05_bit_manipulation/problem_05_06_conversion.cpp
#include "problem_05_06_conversion.h"

namespace chapter_05{
    int conversion(int A, int B){
        int count = 0;
        while(A > 0 || B > 0){
            if(A & 1){
                if (!(B & 1)){
                    count ++;
                }
            } else {
                if (B & 1){
                    count ++;
                }
            }
            A >>= 1;
            B >>= 1;
        }
        return count;
    }
}
